app/com/linkedin/drelephant/tuning/TuningHelper.java

```diff
@@ -122,4 +133,52 @@ public static void updateJobExecution(JobExecution jobExecution) {
       logger.error("Exception updating job execution " + jobExecution.id, e);
     }
   }
+
+  public static Double getContainerSize(Double memory) {
+    return Math.ceil(memory / ONE_GB) * ONE_GB;
+  }
+
+  public static Double getHeapSize(Double heapSize) {
+    if (heapSize < MINIMUM_HEAP_SIZE_MB) {
+      heapSize = MINIMUM_HEAP_SIZE_MB * 1.0;
+    }
+    return heapSize;
+  }
+
+  public static Double parseMaxHeapSizeInMB(String data) {
+    try {
+      logger.debug(" Heap data " + data);
+      Pattern pattern = Pattern.compile(PATTERN_FOR_MEMORY);
+      Matcher m = pattern.matcher(data);
+      String maxHeapSize;
+      if (m.find()) {
+        maxHeapSize = m.group().trim();
+        maxHeapSize = maxHeapSize.substring(4);
```

Comment:
Can use `group` to get the heap size string. e.g if Pattern = `.*-Xmx([\\d]+)([bBmMgG]).*` then `group(2)` will e your heap size string.

Reply:
I am not sure , what the benefit of the same , is it less time complex ?